### PR TITLE
SAAS-1071: Add delete modal

### PR DESCRIPTION
**Pull request by MaksimMalakhovSimbirsoft**
black
m/107244329/188432742-b4e51058-93ed-4411-ac08-0a26f1e
white
dbeb7.png);

- added tests;

### CODE REVIEW

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx [hunk 1]
```diff
@@ -17,36 +17,63 @@ export const OrganizationDeleteModal: FC<OrganizationDeleteModalProps> = ({
...
+        <p className={styles.confirmMessage}>{Messages.confirmDeletionTitle}</p>
+        <Form onSubmit={onSubmit}>
+          {({ handleSubmit }: FormRenderProps) => (
+            <form
+              onSubmit={handleSubmit}
+              className={styles.deleteForm}
+              data-testid="delete-organization-form"
+            >
+              <div className={styles.inputMargin}>
+                <Input type="text" width={35} onInput={handleInput} />
```
Comment (fabio-silva) [inline src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx:50]: Let's use @percona/platform-core `TextInputField`, it's made to work with react-final-form. No need to use handlers here or useState for the value. Besides `handleSubmit`, render props also contain `values`, and that's an object `{[key: string]: any}` with the form's values, where `key` is the field name (e.g. orgName).
So, for those buttons, you just need to do something like `disabled={values.orgName !== orgName}`

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx [hunk 2]
```diff
@@ -17,36 +17,63 @@ export const OrganizationDeleteModal: FC<OrganizationDeleteModalProps> = ({
...
+                >
+                  {Messages.cancel}
+                </Button>
+                <LoaderButton
+                  data-testid="delete-organization-submit-button"
+                  className={styles.saveButton}
+                  type="submit"
+                  loading={pending}
+                  variant={isDisabledButtonVariant}
+                  disabled={!(inputValue === orgName)}
```
Comment (fabio-silva) [inline src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx:67]:
```suggestion
                  disabled={inputValue !== orgName}
```

**Review by fabio-silva — CHANGES_REQUESTED** [review]

**Comment by fabio-silva** [comment]
Two questions about the design:
- Is orange the color chosen by UI/UX?
- The text is starting way after the warning, in terms of padding. I don't know if there's a mock for this, but I guess the text should be left-aligned with the warning

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.messages.ts [hunk 3]
```diff
@@ -1,6 +1,12 @@
...
   cancel: 'Cancel',
   confirm: 'Confirm',
-  deleteOrganization: (orgName: string) => `Are you sure you want to remove the organization ${orgName}?`,
+  deleteOrganization: (orgName: string) => `Are you sure you want to delete the "${orgName}" organization? 
+  You will not be able to recover it later, and you will permanently lose all the data associated with it.
+  This includes active users, PMM connections and available advisors.`,
+  confirmDeletionTitle: 'To confirm this action, type the name of your organization below:',
   deleteOrganizationTitle: 'Delete Organization',
+  deleteOrganizationButton: 'Delete Organization',
+  deleteOrganizationWarningLabel:
```
Comment (TomasBlazek3PG) [inline src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.messages.ts:10]: DO we need to put the "deleteOrganization" in front of everything if we are in OrganizationDeleteModal?

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts [hunk 4]
```diff
@@ -15,7 +14,41 @@ export const getStyles = ({ spacing }: GrafanaTheme) => ({
   `,
   deleteMessage: css`
     margin-bottom: 0;
-    text-align: center;
+    font-weight: bold;
+    line-height: 20px;
```
Comment (TomasBlazek3PG) [inline src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts:18]: This is due to no similar option in themes typography?

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts [hunk 5]
```diff
@@ -15,7 +14,41 @@ export const getStyles = ({ spacing }: GrafanaTheme) => ({
   `,
   deleteMessage: css`
     margin-bottom: 0;
-    text-align: center;
+    font-weight: bold;
```
Comment (TomasBlazek3PG) [inline src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts:17]: Please use themes typography fontWeight

Comment (TomasBlazek3PG) [reply src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts:17]: The same applies to the whole file

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts (hunk 4 above)
Comment (TomasBlazek3PG) [reply src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts:18]: The same applies to the whole file

**Review by TomasBlazek3PG — CHANGES_REQUESTED** [review]
Please check it if it works in dark theme as well.

Needs tests!

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx [hunk 6]
```diff
@@ -18,35 +18,58 @@ export const OrganizationDeleteModal: FC<OrganizationDeleteModalProps> = ({
...
+                >
+                  {Messages.cancel}
+                </Button>
+                <LoaderButton
+                  data-testid="delete-organization-submit-button"
+                  className={styles.saveButton}
+                  type="submit"
+                  loading={pending}
+                  variant={isDisabledButtonVariant(values[orgName])}
+                  disabled={values[orgName] !== orgName}
```
Comment (fabio-silva) [inline src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx:63]: The input should be like this:
`<TextInputField name="orgName" fieldClassName={styles.inputOrg} />`

Then:
`variant={values.orgName === orgName ? "primary" : "secondary"}`
`disabled={values.orgName !== orgName}`
`buttonClassName={values.orgName === orgName ? "styles.activeColorButton" : "styles.disabledColorButton"}`

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx (hunk 6 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx:63]: Done

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts [hunk 7]
```diff
@@ -15,7 +14,45 @@ export const getStyles = ({ spacing }: GrafanaTheme) => ({
...
+  confirmMessage: css`
+    margin-bottom: 0;
+    font-weight: ${typography.weight.bold};
+  `,
+  marginButtons: css`
+    margin-top: 20px;
+  `,
+  formMargin: css`
+    margin-left: 55px;
+    margin-top: 32px;
```
Comment (fabio-silva) [inline src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts:29]: I see many hardcoded margins, paddings, etc. What about the values from theme?

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts (hunk 7 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.styles.ts]: Done

File: src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx (hunk 6 above)
Comment (fabio-silva) [reply src/pages/ManageOrganization/OrganizationView/OrganizationDeleteModal/OrganizationDeleteModal.tsx:58]: @MaksimMalakhovSimbirsoft This fix wasn't done

**Review by fabio-silva — CHANGES_REQUESTED** [review]

**Review by TomasBlazek3PG — APPROVED** [review]

**Pull request merged**